internal/k8s/k8s.go

```diff
@@ -115,15 +116,6 @@ func New(cfg *Config) (*Client, error) {
 		return nil, fmt.Errorf("creating Kubernetes client: %s", err)
 	}
 
-	namespace := cfg.ConfigMapNS
-	if namespace == "" {
-		bs, err := ioutil.ReadFile("/var/run/secrets/kubernetes.io/serviceaccount/namespace")
-		if err != nil {
-			return nil, fmt.Errorf("getting namespace from pod service account data: %s", err)
-		}
-		namespace = string(bs)
-	}
-
```

Comment:
Why was this removed?  I believe this means that previously, this would work even if no namespace was provided, but now we have a hard requirement on it being provided through the manifest.  Is there any harm in leaving this code in place?

Reply:
I can move this code if you prefer, I personally prefer explicit conf in the manifest that's all

Reply:
ok moved it, not tested but it should work, please have a new look

Reply:
looks ok, but would like to hear test confirmation first.  thanks

Reply:
actually this was easy to test with dev-env since you also removed setting the env var from the manifest